src/test/java/org/jabref/logic/exporter/BibTeXMLExporterTestFiles.java

```diff
@@ -46,33 +42,31 @@
     public BibTeXMLExporter bibtexmlExportFormat;
     public BibtexImporter testImporter;
 
-    @Parameter
-    public String filename;
-    public Path resourceDir;
+    private static Path resourceDir;
 
-    @Rule
-    public TemporaryFolder testFolder = new TemporaryFolder();
+    public static Stream<String> fileNames() throws IOException, URISyntaxException {
+        resourceDir = Paths.get(MSBibExportFormatTestFiles.class.getResource("BibTeXMLExporterTestArticle.bib").toURI()).getParent();
+        System.out.println(resourceDir);
 
-    @Parameters(name = "{0}")
-    public static Collection<String> fileNames() throws IOException, URISyntaxException {
-        try (Stream<Path> stream = Files.list(Paths.get(BibTeXMLExporterTestFiles.class.getResource("").toURI()))) {
+        try (Stream<Path> stream = Files.list(resourceDir)) {
             return stream.map(n -> n.getFileName().toString()).filter(n -> n.endsWith(".bib"))
-                    .filter(n -> n.startsWith("BibTeXML")).collect(Collectors.toList());
+                    .filter(n -> n.startsWith("BibTeXML")).collect(Collectors.toList()).stream();
         }
     }
 
-    @Before
-    public void setUp() throws Exception {
-        resourceDir = Paths.get(BibTeXMLExporterTestFiles.class.getResource("").toURI());
+    @BeforeEach
+    public void setUp(@TempDirectory.TempDir Path testFolder) throws Exception {
         databaseContext = new BibDatabaseContext();
         charset = StandardCharsets.UTF_8;
         bibtexmlExportFormat = new BibTeXMLExporter();
-        tempFile = testFolder.newFile();
+        Files.createFile(testFolder.resolve("ARandomlyNamedFile.tmp"));
+        tempFile = testFolder.resolve("ARandomlyNamedFile.tmp").toFile();
         testImporter = new BibtexImporter(mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS), new DummyFileUpdateMonitor());
     }
 
-    @Test
-    public final void testPerformExport() throws IOException, SaveException {
+    @ParameterizedTest
+    @MethodSource("fileNames")
+    public final void testPerformExport(String filename) throws IOException, SaveException {
         String xmlFileName = filename.replace(".bib", ".xml");
         Path importFile = resourceDir.resolve(filename);
         String tempFilename = tempFile.getCanonicalPath();
```

Comment:
Path has a method getFileName which you should use here

Reply:
I'm afraid that is a naming error. The `tempFilename` doesn't represent "filename" but "an absolute path". And you can see that in the pic below. I print `tempFilename` to the console.
![image](https://user-images.githubusercontent.com/25541992/44153279-64aa0748-a0da-11e8-9746-71187aaff9f7.png)
So I guess if it is better to rename this variable to something like `tempFilePath` ? And I've done that in the commit.